### ISSUE TITLE
tests: go test with -race flag taking more than 10 min under Windows. Check for possible anomalies.

### ISSUE THREAD

**Issue by vadmeste** [minio/minio#3590]
## Expected Behavior
We need to see why tests are taking more than 10 min under Windows antrid judge if this is a bug or we just have many test

## Current Behavior
go test -race currently raises timeout because tests are taking more than 10 min

## Possible Solution
Not yet

## Steps to Reproduce (for bugs)
Check appveyor log for time taken by go test

## Context
Manual testing

## Your Environment
Windows

**Comment by harshavardhana** [comment]
Should be fixed by #3712

**Comment by harshavardhana** [comment]
Current CI runs have been reduced to with in < 15mins, we still need to investigate long running tests for windows and reduce their value further.  Of course not sacrificing coverage.

**Comment by kannappanr** [comment]
Over time this has been reduced and the unit test coverage has also increased. Currently, both Travis and Appveyor are taking about 12 minutes and 20 seconds to finish. Closing the issue as we have reduced the time and also the time taken seems to be consistent with what we see in a linux system. 

Here are sample builds for reference
https://travis-ci.org/minio/minio/builds/329718119?utm_source=github_status&utm_medium=notification
https://ci.appveyor.com/project/Harshavardhana/minio-qxbjq/build/11504
